.github/workflows/on-pr-debug.yml

```diff
@@ -77,10 +77,83 @@ jobs:
         env:
           VCPKG_BINARY_SOURCES: "clear;x-gha,readwrite"
         run: |
-          cmake -B build -S . -DCMAKE_C_FLAGS=/W4 -DNATS_BUILD_STREAMING=OFF
+          cmake -B build -S . -DCMAKE_C_FLAGS=/W4
           cmake --build build --config Debug
 
       - name: Test
+        shell: bash
```

Comment:
0/5 as a subsequent improvement (separate PR), maybe consolidate this with build-test.yml, so we have a windows and a unix versions, and keep the on-... workflow files shorter/simpler.